src/helpers/randomizeCard/index.js

```diff
@@ -0,0 +1,736 @@
+			this.effect = this.effect.replace("On play","{pirateCondPlay}")
+		} else {
+			this.effect = this.effect.replace("On play","{condPlay}")
+		}
+			
+		}
+		if ((this.effect.includes("Before attacking")) && randint(1,3) == 1) {
+			this.effect = this.effect.replace("Before attacking","{condAttack}")
+		}
+		
+		this.fillSlots()
+		
+		// Unique cases:
+		if ((this.effect.includes(" it")) && (this.effect.includes("units")) && !(this.effect.includes("itself"))) {
+			this.effect = this.effect.replaceAll(" it"," them")
+		}
+		if (this.effect.includes("poison ") && this.effect.includes("poisoned")) {
+			this.effect = this.effect.replaceAll(" poisoned "," ")
+		}
+		if (this.effect.includes("freeze ") && this.effect.includes("frozen")) {
+			this.effect = this.effect.replaceAll(" frozen "," ")
+		}
+		if ((this.tribe.name == "construct") && (this.effect.includes("unit")) && (randint(1,2) == 1)) {
+			this.effect = this.effect.slice(0,this.effect.indexOf(",")) + this.effect.slice(this.effect.indexOf(",")).replaceAll("unit","Construct")
+			this.effCost *= 0.5
+			if (verbose) print(`Adding Construct-only multiplier of 0.5. New cost: ${this.effCost}`)
+		}
+		if ((this.tribe.name == "satyr") && (this.effect.includes("unit")) && (randint(1,3) != 1)) {
+			this.effect = this.effect.slice(0,this.effect.indexOf(",")) + this.effect.slice(this.effect.indexOf(",")).replaceAll("unit","Satyr")
+			this.effCost *= 0.5
+			if (verbose) print(`Adding Satyr-only multiplier of 0.5. New cost: ${this.effCost}`)
+		}
+		
+		this.rarity = "Epic"
+		if (this.effCost%1 < 0.66) {
+			this.rarity = "Rare"
+		}
+		if (this.effCost%1 < 0.33) {
+			this.rarity = "Common"
+		}
+	}
+
+	fillSlots() {
+		while(true) {
+			var foundState = 0
+			var target = ""
+			var char = ""
+			for (let i=0; i<this.effect.length; i++) {
+				char = this.effect[i]
+				if (char == '{') {
+					foundState = 1
+				}
+				if (foundState == 1 && char != '{' && char != '}') {
+					target += char
+				}
+				if (char == '}' && foundState == 1) {
+					foundState = 2
+					break
+				}
+			}
+			
+			if (foundState != 2) {return}
+
+			var valid = []
+			slots.forEach((slot) => {
+				if (slot.type == target) {valid.push(slot)}
+			})
+				
+			if (this.faction == "Shadowfen") {
+				slotsShadowfen.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Ironclad") {
+				slotsIronclad.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Winter") {
+				slotsWinter.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			if (this.faction == "Swarm") {
+				slotsSwarm.forEach((slot) => {
+					if (slot.type == target) {valid.push(slot)}
+				})
+			}
+			
+			var targetSlot = rchoice(valid)
+			
+			this.effect = this.effect.replace("{"+target+"}",targetSlot.part1)
+			this.effect = this.effect.replace("{"+target+"2}",targetSlot.part2)
```

Comment:
I think we can probably make this a little more elegant in the future just so it’s less manual to replace tokens and construct the whole ability. I’ll play with it once it‘s merged. :)